README.md

```diff
@@ -11,12 +11,25 @@ jobs:
   steps:
     ...
     - name: Upload wheel
-      uses: scientific-python/upload-nightly-action@main
+      uses: scientific-python/upload-nightly-action@8f0394fd2aa0c85d7364a9958652e8994e06b23c # 0.1.0
```

Comment:
could this be a major version instead, the same we do for checkout, or python, or etc?

Reply:
It depends on how much you trust the project; releases / tags can be removed and replaced, while SHAs are a bit harder to fake. But if you trust official releases made by the org, and you just want to review the new action to make sure nothing big changed, versions are fine.

Reply:
well, I would rather like to think that the releases made in the `scientific-python` org are official and trustworthy :)

Reply:
(after all, we all use a lot of `main` actions, e.g. the artifact upload one, which is I agree is not super ideal as a best practice)

Reply:
In that case, if you trust that bunch, versions seem fine ;)

Reply:
I think that @ksunden gave a good and concise summary in https://github.com/matplotlib/matplotlib/pull/26023#discussion_r1212539700. This is not about trusting orgs or not trusting orgs. It is about trying to be extra security cautious for anything that is publishing distributions to any package index. Is this overkill? Yeah, for sure in my mind, but it also seems like a pattern with no downsides other than you have to have your eyes track a little further across the screen to read the associated tag.

> after all, we all use a lot of main actions, e.g. the artifact upload one, which is I agree is not super ideal as a best practice

I will fully agree with you though that tags are better than `main`, and patch version tags are better than major version tags. Though I'm also willing to agree that there should be additional security hardening than using a `latest` tag Docker image

https://github.com/scientific-python/upload-nightly-action/blob/dbd5a4da48dfa98ef9f67c571248ea4891c651fd/Dockerfile#L1

and so using the SHAs only really helps later releases (`0.1.0` is the first commit in the repo).

Reply:
Thanks for the detailed reply. To make this bikeshed really shiny, could I ask you to include some of Kyle's concise commentary as a comment in this readme? Or even just the link you have above would be useful.